scripts/build-spidermonkey.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+set -x
+
+# If WASI_SDK_PATH is set, use it to set up the compiler environment.
+if [[ -n "${WASI_SDK_PATH:-}" ]]; then
+  export CC="${WASI_SDK_PATH}/bin/clang"
+  export CXX="${WASI_SDK_PATH}/bin/clang++"
+  if [[ -z "${HOST_CC:-}" ]]; then
+    export HOST_CC=$(which clang)
+  fi
+  if [[ -z "${HOST_CXX:-}" ]]; then
+    export HOST_CXX=$(which clang++)
+  fi
+else
+  # Otherwise, check that all required environment variables are set.
+  if [[ -z "${WASI_SYSROOT:-}" ]]; then
+      echo "Error: WASI_SYSROOT environment variable is not set."
+      exit 1
+  fi
+  if [[ -z "${CC:-}" ]]; then
+      echo "Error: CC environment variable must be set to a clang that can target wasm32-wasip1."
+      exit 1
+  fi
+  if [[ -z "${CXX:-}" ]]; then
+      echo "Error: CXX environment variable must be set to a clang++ that can target wasm32-wasip1."
+      exit 1
+  fi
+  if [[ -z "${HOST_CC:-}" ]]; then
+      echo "Error: HOST_CC environment variable is not set."
+      exit 1
+  fi
+  if [[ -z "${HOST_CXX:-}" ]]; then
+      echo "Error: HOST_CXX environment variable is not set."
+      exit 1
+  fi
+fi
+
+mode="${1:-release}"
+weval=""
+if [[ $# -gt 1 ]] && [[ "$2" == "weval" ]]; then
+    weval=-weval
+fi
+mozconfig="$(pwd)/mozconfig-${mode}${weval}"
+objdir="obj-$mode${weval}"
+outdir="$mode${weval}"
+
+cat << EOF > "$mozconfig"
+ac_add_options --enable-project=js
```

Comment:
Should we enable `lto`?  `--enable-lto=<full | thin>`

Reply:
Good call: I did that. Makes linking take a lot longer, but that seems good, if anything :)